pkg/controller/master/virtualmachine/vm_volume_status_controller.go

```diff
@@ -0,0 +1,493 @@
+					logrus.Infof("pvc:(%s/%s), fail to update volume:%s status:%s, error:%s", pvcNamespace, pvcName, volName, status.String(), fmt.Errorf("%w", err))
+				}
+				return err
+			}
+
+		}
+	}
+
+	return nil
+}
+
+// LH volume is deleted, remove the possible volue status from PVC
+func (h *VMVolumeStatusController) removeLHVolumeStatusFromPVC(volName string, pvcNamespace string, pvcName string) error {
+	var err error
+	var pvc *corev1.PersistentVolumeClaim
+	if pvc, err = h.getPVC(pvcNamespace, pvcName); err != nil {
+		if !errors.IsNotFound(err) {
+			logrus.Infof("fail to get owner pvc:(%s/%s), give up to remove status of LH volume:%s, error:%s", pvcNamespace, pvcName, volName, fmt.Errorf("%w", err))
+		}
+		return nil
+	}
+
+	if _, ok := pvc.Annotations[util.AnnotationVolumeStatus]; ok {
+		if _, err := h.removePVCAnnotationsVolumeStatus(pvc); err != nil {
+			if !errors.IsConflict(err) {
+				logrus.Infof("pvc:(%s/%s), fail to remove LH volume:%s status, error:%s", pvcNamespace, pvcName, volName, fmt.Errorf("%w", err))
+			}
+			return err
+		}
+	}
+
+	return nil
+}
+
+// as per: func (o AnnotationSchemaOwners) List(ownerGK schema.GroupKind) []string
+func (h *VMVolumeStatusController) getPVCRefOwnerVMs(pvc *corev1.PersistentVolumeClaim) ([]string, error) {
+	if pvcOwners, err := ref.GetSchemaOwnersFromAnnotation(pvc); err != nil {
+		return nil, err
+	} else if len(pvcOwners) == 0 {
+		// pvc is not attched to any owner yet
+		return nil, nil
+	} else {
+		vmGVK := kubevirtapis.VirtualMachineGroupVersionKind
+		vmGK := vmGVK.GroupKind()
+		refOwnerVms := pvcOwners.List(vmGK)
+		return refOwnerVms, nil
+	}
+}
+
+// pvc annotation is further updated to VM annotation
+func (h *VMVolumeStatusController) OnPVCChange(key string, pvc *corev1.PersistentVolumeClaim) (*corev1.PersistentVolumeClaim, error) {
+	if pvc == nil || pvc.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	var err error
+	// process "pvc is detached vm" first, pvc may be updated after this call, following needs to return pvc
+	if pvc, err = h.removeVolumeStatusFromDetachedVM(pvc); err != nil {
+		return nil, err
+	}
+
+	var refOwnerVms []string
+	if refOwnerVms, err = h.getPVCRefOwnerVMs(pvc); err != nil {
+		return pvc, err
+	} else if len(refOwnerVms) == 0 {
+		// pvc is not attched to any owner yet
+		return pvc, nil
+	}
+
+	update := false
+	var volStatus string
+	var ok bool
+	// when pvc has !empty volume annotations, then update(add) to vm, otherwise remove from vm
+	if volStatus, ok = pvc.Annotations[util.AnnotationVolumeStatus]; ok && volStatus != "" {
+		update = true
+	}
+
+	for _, owner := range refOwnerVms {
+		namespace, vmName := ref.Parse(owner)
+		var err error
+		var vm *kubevirtapis.VirtualMachine
+		if vm, err = h.getVM(namespace, vmName); err != nil {
+			if !errors.IsNotFound(err) {
+				logrus.Infof("fail to get owner vm:(%s/%s), give up to update pvc status:%s, error:%s", namespace, vmName, volStatus, fmt.Errorf("%w", err))
```

Comment:
From my experience, formatting with `%w` in `logrus.Infof` works. (Other places need similar updates as well)

```suggestion
				logrus.Infof("fail to get owner vm:(%s/%s), give up to update pvc status:%s, error:%w", namespace, vmName, volStatus, err)
```

Reply:
`logrus does not support %w`, it will print something like 

time="2022-02-11T10:44:51Z" level=error msg="fail to get pvc:(default/vol-600g), give up to map LH volume:pvc-7717ffe7-f610-4496-aa0a-bc5c6feb04ff status:insufficient storage, error:`%!w(*fmt.wrapError=&{fail to get pvc(default/vol-600g), persistentvolumeclaims \"vol-600g\" not found 0xc003334960})"`